include/miopen/miopen.h

```diff
@@ -4704,6 +4706,231 @@ miopenReduceTensor(miopenHandle_t handle,
+
+/*! @brief Sets a tensor descriptor for the specified argument.
+ *
+ * @param problem    Problem to update
+ * @param id         Id of the argument for the descriptor
+ * @param descriptor Tensor descriptor to set
+ * @return           miopenStatus_t
+ */
+miopenStatus_t miopenSetProblemTensorDescriptor(miopenProblem_t problem,
+                                                miopenTensorArgumentId_t id,
+                                                const miopenTensorDescriptor_t descriptor);
+
+/*! @brief The miopenFindOptions allows the user to configure how find will be used.
+ */
+MIOPEN_DECLARE_OBJECT(miopenFindOptions);
+
+/*! @brief Initializes miopenFindOptions object.
+ *
+ * @param options    Pointer to options object to initialze
+ * @return           miopenStatus_t
+ */
+miopenStatus_t miopenCreateFindOptions(miopenFindOptions_t* options);
+
+/*! @brief Destroys miopenFindOptions object.
+ *
+ * @param options    Options object to destroy
+ * @return           miopenStatus_t
+ */
+miopenStatus_t miopenDestroyFindOptions(miopenFindOptions_t options);
+
+/*! @brief Sets the tuning find option. Default value is zero.
+ *
+ * @param options    Options object to upfate
+ * @param value      Value of zero means no tuning, value of one means tuning enabled
+ * @return           miopenStatus_t
+ */
+miopenStatus_t miopenSetFindOptionTuning(miopenFindOptions_t options, int value);
+
+/*! @brief Sets the results order find option. Default value is miopenFindResultsOrderByTime.
+ *
+ * @param options    Options object to upfate
+ * @param value      Specifies what order should find results have
+ * @return           miopenStatus_t
+ */
+miopenStatus_t miopenSetFindOptionResultsOrder(miopenFindOptions_t options,
+                                               miopenFindResultsOrder_t value);
+
+/*! @brief Sets the workspace limit find option. Default value is maximum of size_t
+ *
+ * @param options    Options object to upfate
+ * @param value      Specifies the workspace limit for find call. All solvers exceeding the limit
+ * would be ignored.
+ * @return           miopenStatus_t
+ */
+miopenStatus_t miopenSetFindOptionWorkspaceLimit(miopenFindOptions_t options, size_t value);
+
+/*! @brief The miopenSolution object describes a prepared solution.
+ */
+MIOPEN_DECLARE_OBJECT(miopenSolution);
+
+/*! @brief Finds solutions to a problem by running different applicable solutions. Memory is
+ * automatically allocated.
+ *
+ * @param handle       Handle to execute the kernels
+ * @param problem      Problem to solve
+ * @param options      Find options. When null default values would be used
+ * @param solutions    Pointer to the first result. Must not be null
+ * @param numSolutions Pointer to the amount of results. Ignored if null
+ * @param maxSolutions Limits the amount of results
+ * @return             miopenStatus_t
+ */
+miopenStatus_t miopenFindSolutions(miopenHandle_t handle,
```

Comment:
@DrizztDoUrden @umangyadav Do we have a ticket intended for discussion of Find 2.0 API? Current API enforces the library to allocate buffers for tensors. I am afraid this approach is incompatible with frameworks that do manage GPU memory themselves, and may lead to memory starvation issues with really big tensors/networks.

Reply:
@atamazov We could allow passing pre-allocated tensors via find options. That would only require one new universal API method and would keep the API flexible.

Reply:
memory starvation looks like a real issue. I wasn't planning on opening a dedicated ticket for find 2.0. Do we have any existing ticket for find-2.0 that we can re-use?

Reply:
I think separate issues should be confined to their own tickets. We can either tag them with some label or create an umbrella ticket with a list of other issues.